julearn/pipeline/tests/test_pipeline_creator.py

```diff
@@ -158,12 +160,113 @@ def test_hyperparameter_tuning(
     kind = "grid"
     if search_params is not None:
         kind = search_params.get("kind", "grid")
+
     if kind == "grid":
         assert isinstance(pipeline, GridSearchCV)
         assert pipeline.param_grid == param_grid
-    else:
+    elif kind == "random":
         assert isinstance(pipeline, RandomizedSearchCV)
         assert pipeline.param_distributions == param_grid
+    elif kind == "bayes":
+        assert isinstance(pipeline, BayesSearchCV)
+        assert pipeline.search_spaces == param_grid
+
+
+def _compare_param_grids(a, b):
+    """Compare two param grids.
+
+    Parameters
+    ----------
+    a : dict
+        The first param grid (processed).
+    b : dict
+        The second param grid (raw).
+
+    """
+    for key, val in a.items():
+        assert key in b
+        if hasattr(val, "rvs"):
+            assert val.args[0] == b[key][0]
+            assert val.args[1] == b[key][1]
+            if b[key][2] in ["log-uniform", "loguniform"]:
+                assert val.dist.name == "loguniform"
+            elif b[key][2] == "uniform":
+                assert val.dist.name == "uniform"
+            else:
+                raise ValueError(f"Unknown distribution {b[key][2]}")
+        else:
+            assert val == b[key]
+
+
+def test_hyperparameter_tuning_distributions(
+    X_types_iris: Dict[str, List[str]],  # noqa: N803
+    model: str,
+    preprocess: Union[str, List[str]],
+    problem_type: str,
+    get_tuning_distributions: Callable,
+    search_params: Dict[str, List],
+) -> None:
+    """Test that the pipeline hyperparameter tuning works as expected.
+
+    Parameters
+    ----------
+    X_types_iris : dict
+        The iris dataset features types.
+    model : str
+        The model to test.
+    preprocess : str or list of str
+        The preprocessing steps to test.
+    problem_type : str
+        The problem type to test.
+    get_tuning_distributions : Callable
+        A function that returns the tuning hyperparameters for a given step.
+    search_params : dict of str and list
+        The parameters for the search.
+
+    """
+    from skopt import BayesSearchCV
```

Comment:
This test will fail if the dependency is missing, is there a check for it?

Reply:
I'm actually tempted to have a test only for the BayesSearchCV. Now my problem is that the test is exactly the same as the others, it's just that the fixture changes. Does it make sense to encapsulate the test in a private function and then call it with two fixtures?

Reply:
Not sure I follow you there, will it be possible to have a small example snippet?

Reply:
Like I did not